YAFCparser/Data/FactorioDataDeserializer.cs

```diff
@@ -84,7 +84,15 @@ public Project LoadData(string projectPath, LuaTable data, LuaTable prototypes,
             raw = (LuaTable)data["raw"];
             foreach (var prototypeName in ((LuaTable)prototypes["item"]).ObjectElements.Keys)
                 DeserializePrototypes(raw, (string)prototypeName, DeserializeItem, progress);
-            recipeModules.SealAndDeduplicate(universalModules.ToArray());
+
+            Item[] universalModulesArray = universalModules.ToArray();
+            IEnumerable<Item> FilteredModules(Recipe item)
+            {
+                Func<Item, bool> AllowedModulesFilter(Recipe key) => item => !item.module.limitation_blacklist.Contains(key);
```

Comment:
`limitation_blacklist` can be null, leading to exceptions
load vanilla to reproduce this case

Reply:
Nice catch, it has been a while since I played vanilla Factorio, so it did even cross my mind that this could cause issues.